File: PathfinderAPI/src/main/java/org/bukkit/pathfinding/ControllerMove.java
package org.bukkit.pathfinding;

public interface ControllerMove {

    boolean isOperationMove();

    void move(double motionX, double motionY, double motionZ, double speed);

    void move(float forward, float sideward);

    void update();


    double getX();

    double getY();

    double getZ();

    double getSpeed();


}

File: PathfinderAPI/src/main/java/com/github/ysl3000/bukkit/pathfinding/pathfinding/ControllerMove.kt
package com.github.ysl3000.bukkit.pathfinding.pathfinding

@Deprecated("")
interface ControllerMove {
    fun move(motionX: Double, motionY: Double, motionZ: Double, speed: Double)
    fun move(forward: Float, sideward: Float)
    fun update()
    fun getX(): Double
    fun getY(): Double
    fun getZ(): Double
    fun getSpeed(): Double
    fun isOperationMove(): Boolean
}
